architecting-cloud-native-dotnet-apps-for-azure/cloud-native-devops.md

```diff
@@ -5,8 +5,246 @@ ms.date: 06/30/2019
+
+The end result of a build is a collection of files known as build artifacts. These can be passed along to the next step in the build process or added to an Azure Artifact feed so they can be consumed by other builds. 
+
+### Azure DevOps Releases
+
+Builds take care of compiling the software into a shippable package but the artifacts still need to be pushed out to a testing environment to complete continuous delivery. For this Azure DevOps uses a separate tool called Releases. Releases make use of the same library of tasks which were available to the Build but introduce a concept of "stages". A stage is an isolated environment into which the package is installed. For instance a product might make use of a develop, a QA and a production environment. Code is continuously delivered into the development environment where automated tests can be run against it. Once those tests pass the release moves onto the QA environment for manual testing. Finally the code is pushed to production where it is visible to everybody. 
+
+![An example release pipeline with Develop, QA and Production phases](media/release-pipeline.png)
+
+Each stage in the build can be automatically trigged by the completion of the previous phase. In many cases, however, this isn't desirable. Moving code into production might require approval from somebody. Releases supports this by allowing approvers at each step of the release pipeline. Rules can be set up such that a specific person or group of people must sign off on a release before it makes it to production. These gates allow for manual quality checks and also for compliance with any regulatory requirements pertaining to controls on what goes into production. 
+
+### Everybody Gets a Build Pipeline
+
+There is no cost to configuring many build pipelines so it is advantageous to have at least one build pipeline per microservice. Ideally microservices are independently deployable to any environment so having each one able to be released via its own pipeline without releasing a mass of unrelated code is perfect. Each pipeline can have its own set of approvals allowing for variations in build process for each service. 
+
+### Versioning Releases
+
+One drawback to using the Releses functionality is that it cannot be defined in a checked in `azure-pipelines.yml` file. There are many reason you might want to do that from having per-branch release definitions to including a release skeleton in your project template. Fortunately work is ongoing to shift some of the stages support into the Build component. This will be known as multi-stage build and the first version are available right now!
```

Comment:
many reason -> many reasons
"are available right now!" -> where? link for more info?